app/controllers/course/discussion/posts_controller.rb

```diff
@@ -0,0 +1,106 @@
+class Course::Discussion::PostsController < Course::ComponentController
+  before_action :set_topic
```

Comment:
Hence delegator pattern. The subclass can just implement the abstract methods... The contract for set_X is more general than X

Simple form does not require instance variables, it can handle both instance variable and methods?